### PR TITLE
✨ Source Chargebee: add new fields in streams #32098

### PR DESCRIPTION
**Pull request by KimPlv**
## What
Add new fields: 

**addon**: 
- channel
- object
- type

**coupons**: 
- object

**credit_notes** 
- base_currency_code
- business_entity_id
- channel
- exchange_rate
- is_digital
- is_vat_moss_registered
- object

**customers**: 
- business_entity_id
- channel
- object

**event**: 
- object 

**invoices**:
- base_currency_code

**channel**
- object

**item**:
- external_name

**orders**: 
- base_currency_code
- business_entity_id
- exchange_rate
- object

**payment_source**:
- object

**plan**: 
- channel
- charge_model
- object

**promotional_credits**:
- object 

**subscriptions**: 
- business_entity_id
- channel

**transactions**:

- base_currency_code
- business_entity_id
- error_text
- object
- payment_method_details
- reference_number

## How
update schema folders

### CODE REVIEW

**Review by marcosmarxm — CHANGES_REQUESTED** [review]
Can you bump the metadata, dockerfile and doc changelog version?

**Comment by marcosmarxm** [comment]
Thanks for the contribution @KimPlv is it possible to give me access to your branch? I need to send some updates for expected records.

**Comment by KimPlv** [comment]
> Can you bump the metadata, dockerfile and doc changelog version?

Update for the DockerImageTag in metadata.yaml
Changelog version in doc already increased
No Dockerfile

**Comment by lideke** [comment]
Hi @marcosmarxm,
We are facing more issues with chargebee where some fields are missing.
Right now i'm switching to our build but it will be great if you can accept the PR.
Thanks

**Comment by marcosmarxm** [comment]
Hello @KimPlv sorry the delay to get this merged. This week the codebase is freeze because of holidays. I'll run tests and return to you, thanks for the patience.

**Comment by lideke** [comment]
> Hello @KimPlv sorry the delay to get this merged. This week the codebase is freeze because of holidays. I'll run tests and return to you, thanks for the patience.

Hi @marcosmarxm, Kim she on vacation, i'll follow the PR
Just to let you know we are already using in our side on production for missing fields in chargebee streams
Thanks you to take time to test and validate the PR

**Comment by pnilan** [comment]
@lideke @KimPlv I can't seem to find any record of `is_vat_moss_registered` in the Chargebee API documentation for `credits_notes`. Also does not show up in the sample responses. Can you verify this field for the `credits_notes` stream? Thanks.

File: airbyte-integrations/connectors/source-chargebee/source_chargebee/schemas/credit_note.json [hunk 1]
```diff
@@ -4,453 +4,863 @@
...
+        "null"
+      ]
+    },
+    "object": {
+      "type": [
+        "string",
+        "null"
+      ]
+    },
+    "is_vat_moss_registered": {
```
Comment (pnilan) [inline airbyte-integrations/connectors/source-chargebee/source_chargebee/schemas/credit_note.json:234]: Please verify this field.

**Review by pnilan — CHANGES_REQUESTED** [review]

File: airbyte-integrations/connectors/source-chargebee/source_chargebee/schemas/credit_note.json (hunk 1 above)
Comment (KimPlv) [reply airbyte-integrations/connectors/source-chargebee/source_chargebee/schemas/credit_note.json]: Hi @pnilan, we can collect data for this field, but it depends on the record, in some cases we have it and in others not. That's why i've removed the field from the credit note schema, and added "additionalProperties" to true to include it anyway. It's good for you like this ?

Comment (pnilan) [reply airbyte-integrations/connectors/source-chargebee/source_chargebee/schemas/credit_note.json]: @KimPlv Thanks for the confirmation. I confirmed with Chargebee support that this specific field is only available for accounts that are configured for EU taxes, which is why it wasn't populating on my end. I think we can go ahead and include it in the schema and add a `$comment` property to note this.

I will update this on my end and verify the tests are running properly and then we should be able to merge the PR. I will have to merge this as a separate PR though for our CI pipeline to run properly. I'll let you know when I have the PR submitted.

Comment (KimPlv) [reply airbyte-integrations/connectors/source-chargebee/source_chargebee/schemas/credit_note.json:234]: @pnilan fine ! Thanks for the information. I've made the change for this field

**Pull request closed without merging**